### PR TITLE
[LPS-116504, LPS-112471, LPS-112472] Updates clay:progressbar and clay:radio and adds props attribute to react:component

### PR DESCRIPTION
**Pull request by jbalsas**
I know this has better chances at making it through separated and it would probably be the way to go, but I'm going to give this a go since most of the changes seem trivial...

**In this PR we:**
- Update `clay:progressbar` to use a java-based version instead of the old soy-based one. Checked 1 existing usage
- Update `clay:radio` to use a java-based version instead of the old soy-based one. Checked 1 existing usage
- Adds `props` attribute to `react:component` to better match expectations instead of calling it `data`
- Updates usages of `data` to `props`

**How to test it:**
- The update of `props` in `react:component` should be pretty obvious throughout CI... hoping to see issues there :fingers-crossed:
- Tested `clay:progressbar` in the User Associated Data widget. Check the [Exporting User Data](https://help.liferay.com/hc/en-us/articles/360029132051-Exporting-User-Data) tutorial (won't work with default user Test test, need to create another admin user for it).

Screen Shot 2020-07-20 at 16 18 06

- Tested `clay:radio` in the User Associated Data widget. Check the [Sanitizing User Data](https://help.liferay.com/hc/en-us/articles/360028819072-Sanitizing-User-Data#the-personal-data-erasure-screen) tutorial (won't work with default user Test test, need to create another admin user for it).
Screen Shot 2020-07-20 at 16 16 40

**Not done:**
- Progressbars in the uad example are slightly misaligned... not sure if this was there before or is the result of this PR. Would need a followup
- Many  apis passing `props` to `react:component` are named `*DisplayContext.getSomethingData()`. I haven't renamed them here, but we might wanna for consistency... 🤔

### CODE REVIEW

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
Ok, SF and me cool, no-one overreact to this 😂

File: modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/view_apps.jsp [hunk 1]
```diff
@@ -23,24 +23,24 @@
...
-				"pathFriendlyURLPublic", PortalUtil.getPathFriendlyURLPublic()
-			).put(
-				"userId", themeDisplay.getUserId()
-			).build()
-		%>'
-		module="js/pages/apps/index.es"
-	/>
+		<react:component
+			module="js/pages/apps/index.es"
+			props='<%=
```
Comment (carloslancha) [inline modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/view_apps.jsp:28]: Everywhere else `props` are being defined outside the tag call. Don't we have any SF rule for this? We just don't care about it?

File: modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/layout/details.jsp [hunk 2]
```diff
@@ -103,7 +103,8 @@ String friendlyURLBase = StringPool.BLANK;
 
 				<div class="btn-url-history-wrapper">
 					<react:component
-						data='<%=
+						module="js/friendly_url_history/FriendlyURLHistory"
+						props='<%=
```
Comment (carloslancha) [inline modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/layout/details.jsp:107]: Looks like it's not everywhere, just **almost** everywhere 😛

**Comment by brunobasto** [comment]
ci:test:sf

**Comment by brunobasto** [comment]
Removed generics from deprecation javadoc. SF fails otherwise and it seems to be how we do it in [other places](https://github.com/liferay/liferay-portal/blob/009c90ee86104d8a6886cbb93d34db4bdb36387d/portal-kernel/src/com/liferay/document/library/kernel/service/DLFileEntryTypeServiceWrapper.java#L51)

**Comment by brunobasto** [comment]
ci:test:sf

**Comment by brunobasto** [comment]
Fixed baseline error. Running tests again..

**Comment by brunobasto** [comment]
ci:test:relevant

**Comment by brunobasto** [comment]
ci:test:sf

**Comment by brunobasto** [comment]
Solved conflicts. Retesting..

**Comment by brunobasto** [comment]
ci:test:relevant

**Comment by brunobasto** [comment]
@john-co Do you think those failures are related?

**Comment by john-co** [comment]
- Success message toast looks off in first two testcases, possibly related.
PR:
image
Upstream:
screenshot_19

- SitetemplatesUsecase#SiteTemplatePageLinkInWebContentRedirectCorrectly may be a test fix

@zhangyan0701 can you please investigate further?

**Comment by zhangyan0701** [comment]
ci:test:bundle

**Comment by jbalsas** [comment]
> Success message toast looks off in first two testcases, possibly related.

I don't think that's related... makes more sense this changed in the Notifications PR

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
I couldn't reproduce any of the failures locally, all tests passed with the same bundle. So more like flaky failures.

**Comment by brunobasto** [comment]
ContentPages tests failed again and now we have some more Forms tests failing. Could be unrelated, but maybe @john-co and @PablitoBonito can confirm this before we do another run?

**Comment by john-co** [comment]
Just started reviewing :)

**Comment by john-co** [comment]
@kenjiheigel , there's common failures being reported as unique in the last run, ie: 
- ContentPagesWithEditMode#OnlyCanEditContentWhenSelectContentEdition
- Forms tests

Comparison details:
https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=3&testrayRunIdB=1244097570&testrayRunIdA=1244061267&statusB=2

Also it looks like the other AppBuilder tests are unrelated due to a whole axis node failing to find localhost:8080. https://test-1-12.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/304255/AXIS_VARIABLE=0,label_exp=!master/
```
start-app-server-preserve-liferay-home:
[beanshell] Waiting for server startup message
     [echo] Waiting for http://localhost:8080/web/guest to load
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/jps_output
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/jstack_output
[stopwatch] [run.batch.test.action: 37:38.270 sec]
     [echo] The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:361: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-batch.xml:1530: /opt/dev/projects/github/liferay-portal/build-test-batch.xml:1534: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-tomcat.xml:50: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test-tomcat.xml:62: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test.xml:12638: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test.xml:13003: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test.xml:4120: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test.xml:4256: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal/build-test.xml:5393: No server is available at http://localhost:8080/web/guest
     [echo] 2020-07-23 03:00:10
     [echo] Full thread dump Java HotSpot(TM) 64-Bit Server VM (25.121-b13 mixed mode):
```

**Comment by john-co** [comment]
@manoelcyreno, @Tim-Cao this seems like a flaky test: https://testray.liferay.com/home/-/testray/cases/1230987993 can you please confirm?

**Comment by john-co** [comment]
Reviewed. Test failures are unrelated. :heavy_check_mark: 

Due to sheer number of jobs/tests being run, I recommend no retest is needed and can be manually forwarded if no other code changes are needed.

**Comment by kenjiheigel** [comment]
@john-co , thanks for the heads up. Will take a look and try to see what's going on.

**Comment by jbalsas** [comment]
Ok then... manually forwarded: https://github.com/brianchandotcom/liferay-portal/pull/91748

**Comment by vicnate5** [comment]
@manoelcyreno `UndoPagechangeHundredTimes` sounds like a great test, but seems weird to run this type of test in relevant, it has a least 100 ways to go wrong.

**Pull request closed without merging**